resteasy-cdi/src/main/java/org/jboss/resteasy/cdi/ResteasyCdiExtension.java

```diff
@@ -81,9 +93,50 @@ public static boolean isCDIActive()
    public void observeBeforeBeanDiscovery(@Observes BeforeBeanDiscovery event, BeanManager beanManager)
    {
       this.beanManager = beanManager;
+      final AnnotatedType<ContextProducers> producersAnnotatedType = beanManager.createAnnotatedType(ContextProducers.class);
+      event.addAnnotatedType(producersAnnotatedType, ContextProducers.class.getCanonicalName());
       active = true;
    }
 
+   /**
+    * Process any client beans.
+    *
+    * @param pb the bean being processed
+    */
+   public void processClientBean(@Observes final ProcessBean<Client> pb) {
+      // this method will get notified if there is any bean of type `Client` created
+      final Set<Annotation> qualifiers = pb.getBean()
+              .getQualifiers(); // you want to detect beans with no explicit qualifiers
+      if (qualifiers.contains(Any.Literal.INSTANCE) && qualifiers.contains(Default.Literal.INSTANCE)) {
+         generateClientBean = false;
+      }
+   }
```

Comment:
@manovotn This never seems to be triggered, but the one below does.

Reply:
This is probably because you are defining it as `ProcessBean<Client>`.
`ProcessBean<X>` is a very generic observer that gets notified for any shape of bean - producer method, producer field, class bean or synthetic bean. And `X` stands for the class of the bean which is the issue here; in your case, this is the class which contains the producer method.
Take a look at the javadoc here - https://github.com/eclipse-ee4j/cdi/blob/master/api/src/main/java/jakarta/enterprise/inject/spi/ProcessBean.java#L27-L39

Granted, I didn't realize it either when giving you the code snippet when I first reviewed this PR, so feel free to blame it on me :-)

You could either:
* Have several observer methods, one for each subclass of `ProcessBean` and perform the detection there
  * Has the upside of only being notified about events that you really want to know and is more performance friendly
  * But it is also a lot of code which isn't as nice to read 
* Use just `ProcessBean<?>` and inspect the bean type via something like `pb.getBean().getTypes().contains(Client.java)` to make sure it is the bean you are looking for
  * Easy to write and understand
  * Gets notified about every single bean in the deployment which might create some overhead in massive deployments

In both cases, you'll still want to add the check for `@Any` and `@Default` qualifiers.

Reply:
Okay done. That makes more sense to me. I've updated it to use `ProcessBean<?>` and that seems to be working. If we see deployment overhead we can make some changes.